### PR TITLE
Fix/ Add turnstile token to password reset

### PR DESCRIPTION
**Pull request by xkopenreview**
This pr should add turnstile widget to password reset page
depends on https://github.com/openreview/openreview-api/pull/401

this pr should also extract turnstile widget related logic to a hook so that it can be added to other pages

### CODE REVIEW

File: hooks/useTurnstileToken.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+  useEffect(() => {
+    if (!renderWidget) {
+      setTurnstileToken(null)
+      if (widgetId) {
+        window.turnstile.remove(widgetId)
+      }
+      return
+    }
+    if (window.turnstile) {
+      const id = window.turnstile.render(`#turnstile-${key}`, {
```
[the remark this answers is not in the document]
Comment (zbialecki) [reply hooks/useTurnstileToken.js:17]: Also, you could use the state of the ref to replace the renderWidget prop. If ref is null, that means the container element is not being rendered by the parent.

File: components/FeedbackModal.js [hunk 2]
```diff
@@ -7,20 +7,21 @@ import ErrorAlert from './ErrorAlert'
...
-  const [turnstileToken, setTurnstileToken] = useState(null)
   const [error, setError] = useState(null)
+  const [isOpen, setIsOpen] = useState(false)
   const { accessToken } = useContext(UserContext)
   const [formData, setFormData] = useReducer((state, action) => {
     if (action.type === 'reset') return {}
     if (action.type === 'prefill') return action.payload
     return { ...state, [action.type]: action.payload }
   }, {})
+  const { turnstileToken, turnstileContainerRef } = useTurnstileToken('feedback')
```
Comment (zbialecki) [inline components/FeedbackModal.js:23]: I was actually suggesting the parent component manage the ref and pass it to the hook. Then the parent component could conditionally render the container using some state var. Would this pattern make the code simpler?

File: pages/reset.js [hunk 3]
```diff
@@ -50,8 +52,12 @@ const ResetForm = ({ setEmailSent }) => {
           onChange={(e) => setEmail(e.target.value.trim())}
         />
       </div>
-
-      <button type="submit" className="btn btn-primary" disabled={!isValidEmail(email)}>
+      <div ref={turnstileContainerRef} />
```
Comment (zbialecki) [inline pages/reset.js:55]: Should be an equal margin below this div as there is below the email input

**Review by zbialecki — APPROVED** [review]
Looks like it's working as expected, just a small comment

**Review by melisabok — APPROVED** [review]

**Review by carlosmondra — APPROVED** [review]

**Pull request merged**